### DISCUSSION TITLE
Documentation

### DISCUSSION THREAD

**Discussion by Yogesh-Dubey-Ayesavi** [flutterjanus/flutter_janus_client#148] [Ideas]
It should have it's own dedicated documentation, well we cant directly use the docs of javascript one ?

im thinking to document it using docusauraus

**Comment by shivanshtalwar** [comment]
Well it uses dart docs for documentation already which creates api reference if you want to create a dedicated documentation you are more than welcome to do so

**Reply by Yogesh-Dubey-Ayesavi** [reply]
It's quite messed up even for maintainer's  there are lot's of methods which are not documented well

**Reply by shivanshtalwar0** [reply]
we can start by improving dart docs and then can focus maybe on writing dedicated docs

**Reply by Yogesh-Dubey-Ayesavi** [reply]
May be, i wanted a few insights im not getting

**Reply by shivanshtalwar0** [reply]
insights on what?

**Reply by Yogesh-Dubey-Ayesavi** [reply]
like on janus plugin class why polling timer is used ?

**Reply by shivanshtalwar0** [reply]
simply because janus api supports two general ways of communicating either by rest api for which http polling is must for fetching events from janus and other is websocket i already wrote wiki on it
https://github.com/flutterjanus/flutter_janus_client/wiki

**Reply by Yogesh-Dubey-Ayesavi** [reply]
I am reading janus.js  i will start documenting it sooner.

**Reply by shivanshtalwar0** [reply]
don't assume everything in janus js is implemented as is in this library but core things are more or less same and basic building ideas have been inspired from janusjs only key difference being it encapsulates webrtc stack tightly whereas this library doesn't

**Reply by Yogesh-Dubey-Ayesavi** [reply]
I have made a pr on local stream injection, i request you to have a look at it
